src/extension.ts

```diff
@@ -212,6 +213,34 @@ export async function activate(context) : Promise<extensionapi.ExtensionAPI> {
     }, this);
     await registerYamlSchemaSupport();
 
+
+      const provider: vscode.TextDocumentContentProvider = <vscode.TextDocumentContentProvider>{
```

Comment:
I'd prefer it if we can move this into its own file and extend the vscode.TextDocumentContentProvider interface as needed.